### PR TITLE
use tempfiles to not access file multiple times

### PR DESCRIPTION
**Pull request by ChristianBeilschmidt**
- [x] I agree to follow the project's [code of conduct](https://github.com/georust/gdal/blob/master/CODE_OF_CONDUCT.md).
- [x] I added an entry to `CHANGES.md` if knowledge of this change could be valuable to users.
---

### CODE REVIEW

File: src/programs/raster/mdimtranslate.rs [hunk 1]
```diff
@@ -231,15 +231,55 @@ mod tests {
...
+    /// and returns a tuple of the temp dir (for clean-up) as well as the path to the file.
+    /// We can remove this when <https://github.com/OSGeo/gdal/issues/6253> is resolved.
+    struct TempDataset {
+        _temp_dir: tempfile::TempDir,
+        temp_path: PathBuf,
+    }
+
+    impl TempDataset {
+        pub fn fixture(name: &str) -> Self {
+            let path = std::path::Path::new(file!())
```
Comment (lnicola) [inline src/programs/raster/mdimtranslate.rs:244]: Is this really better than `Path::new("fixtures")`? The tests should run in the root directory of the project.

**Comment by lnicola** [comment]
Can we make a `test_utils` or something to not duplicate that struct?

File: src/programs/raster/mdimtranslate.rs (hunk 1 above)
Comment (ChristianBeilschmidt) [reply src/programs/raster/mdimtranslate.rs:244]: I can revert that, just copied it from fixtures!

**Review by lnicola — APPROVED** [review]

**Comment by jdroenner** [comment]
bors merge

**Pull request merged**